feat(go): refactor options page (#15123)
getsentry/sentry-docs: docs/platforms/go/common/configuration/options.mdx

<PlatformContent includePath="configuration/config-intro" />

## Core Options

Options that can be read from an environment variable (`SENTRY_DSN`, `SENTRY_ENVIRONMENT`, `SENTRY_RELEASE`) are read automatically.

<SdkOption name="Dsn" type="string" envVar="SENTRY_DSN">

> The DSN tells the SDK where to send the events. If this value is not provided, the SDK will try to read it from the `SENTRY_DSN` environment variable. If that variable also does not exist, the SDK will just not send any events.

Reviewer: ```suggestion
The DSN tells the SDK where to send the events. If this value is not provided, the SDK will try to read it from the `SENTRY_DSN` environment variable. If that's also missing, no events will be sent.
```

Learn more about [DSN utilization](/product/sentry-basics/dsn-explainer/#dsn-utilization).

</SdkOption>

<SdkOption name="Debug" type="bool" defaultValue="false">

If debug is enabled, the SDK will attempt to print out useful debugging information if something goes wrong while sending the event. The default is always `false`. It's generally not recommended to turn it on in production, though turning `debug` mode on will not cause any safety concerns.

</SdkOption>

<SdkOption name="Release" type="string" envVar="SENTRY_RELEASE">

> Sets the release. Some Sentry features are built around releases, and, thus, reporting events with a non-empty release improves the product experience. See [the releases documentation](/product/releases/).

Reviewer: ```suggestion
Sets the release. Some Sentry features are built around release info, so reporting releases can help improve the overall experience. See [the releases documentation](/product/releases/).
```

</SdkOption>

<SdkOption name="Environment" type="string" envVar="SENTRY_ENVIRONMENT">

Sets the environment. This string is freeform and not set by default. A release can be associated with more than one environment to separate them in the UI (think `staging` vs `prod` or similar).

By default, the SDK will try to read this value from the `SENTRY_ENVIRONMENT` environment variable.

Environments tell you where an error occurred, whether that's in your production system, your staging server, or elsewhere. Sentry automatically creates an environment when it receives an event with the environment parameter set.

</SdkOption>

<SdkOption name="SampleRate" type="float64" defaultValue="1.0">

Configures the sample rate for error events, in the range of `0.0` to `1.0`. The default is `1.0`, which means that 100% of error events will be sent. If set to `0.1`, only 10% of error events will be sent. Events are picked randomly.

> As a historical special case, the sample rate `0.0` is treated as if it was `1.0`. To drop all events, set the DSN to the empty string.

Reviewer: ```suggestion
As a historical special case, the sample rate `0.0` is treated as if it was `1.0`. To drop all events, set the DSN to an empty string.
```

</SdkOption>

<SdkOption name="MaxBreadcrumbs" type="int" defaultValue="100">

This variable controls the total amount of breadcrumbs that should be captured. However, you should be aware that Sentry has a [maximum payload size](https://develop.sentry.dev/sdk/data-model/envelopes/#size-limits) and any events exceeding that payload size will be dropped.

> When MaxBreadcrumbs is negative then breadcrumbs are ignored.

Reviewer: ```suggestion
If MaxBreadcrumbs is given a negative value, breadcrumbs are ignored.
```

If you enable this option, be sure to manually remove what you don't want to send using our features for managing [_Sensitive Data_](../../data-management/sensitive-data/).

</SdkOption>

<SdkOption name="ServerName" type="string">

This option can be used to supply a server name. When provided, the name of the server is sent along and persisted in the event. For many integrations, the server name actually corresponds to the device hostname, even in situations where the machine is not actually a server.

Most SDKs will attempt to auto-discover this value.

</SdkOption>

<SdkOption name="IgnoreErrors" type="[]string">

A list of regexp strings that will be used to match against event's message and if applicable, caught errors type and value. If the match is found, then a whole event will be dropped.

By default, no errors are ignored.

</SdkOption>

<SdkOption name="MaxErrorDepth" type="int" defaultValue="100">

This is the maximum number of errors reported in a chain of errors. This protects the SDK from an arbitrarily long chain of wrapped errors.

> An additional consideration is that arguably reporting a long chain of errors is of little use when debugging production errors with Sentry. The Sentry UI is not optimized for long chains either. The top-level error together with a stack trace is often the most useful information.

Reviewer: ```suggestion
In practice, reporting very long chains usually provides little value when debugging production issues, as the Sentry UI isn’t optimized for them. The top-level error and its stack trace usually contain the most relevant information.
```

</SdkOption>

## Logging Options

<SdkOption name="EnableLogs" type="bool" defaultValue="false">

> Set this option to `true` to enable log capturing in Sentry. Only when this is enabled will the SDK capture log messages and send them to Sentry.

Reviewer: ```suggestion
Set this option to `true` to enable log capturing in Sentry. The SDK will only capture and send log messages to Sentry if this option is enabled.
```

<SdkOption name="BeforeSend" type="func(event *Event, hint *EventHint) *Event">

This function is called with an SDK-specific message or error event object, and can return a modified event object, or `nil` to skip reporting the event. This can be used, for instance, for manual PII stripping before sending.

By the time `BeforeSend` is executed, all scope data has already been applied to the event. Further modification of the scope won't have any effect.

</SdkOption>

<SdkOption name="BeforeSendTransaction" type="func(event *Event, hint *EventHint) *Event">

This function is called with an SDK-specific transaction event object, and can return a modified transaction event object, or `nil` to skip reporting the event. One way this might be used is for manual PII stripping before sending.

</SdkOption>

<SdkOption name="BeforeBreadcrumb" type="func(breadcrumb *Breadcrumb, hint *BreadcrumbHint) *Breadcrumb">

This function is called with an SDK-specific breadcrumb object before the breadcrumb is added to the scope. When `nil` is returned from the function, the breadcrumb is dropped. To pass the breadcrumb through, return the first argument, which contains the breadcrumb object.

> The callback typically gets a second argument (called a "hint") which contains the original object from which the breadcrumb was created to further customize what the breadcrumb should look like.

Reviewer: ```suggestion
The callback typically gets a second argument (called a "hint") that provides the original object used to create the breadcrumb. You can use this to further customize the breadcrumb’s content or appearance.
```
